packages/lit-dev-content/site/tutorials/content/tooltip/08.md

```diff
@@ -0,0 +1,74 @@
+The tooltip supports good performance via lazy creation, but using the
+`lazy` API is cumbersome. In addition, as implemented, the tooltip content
+doesn't update when the greeted person's name changes.
+
+A Lit directive can address these issues by encapsulating the tooltip's `lazy`
+API. Users of the directive will use an
+[element expression](http://lit.dev/docs/templates/expressions/#element-expressions)
+on the target element and provide a
+[TemplateResult](http://lit.dev/docs/libraries/standalone-templates/#rendering-lit-html-templates)
+containing the tooltip content.
+
+A scaffold for the directive is setup for you to fill in with the `update`
+method already completed. It stores directive `part` and the `tooltipContent`
+passed in so you can use them in the methods you'll add: `setupLazy` and
+`renderTooltipContent`.
+
+```ts
+update(part: ElementPart, [tooltipContent]: DirectiveParameters<this>) {
+  this.tooltipContent = tooltipContent;
+  this.part = part;
+  if (!this.didSetupLazy) {
+    this.setupLazy();
+  }
+  if (this.tooltip) {
+    this.renderTooltipContent();
+  }
+}
+```
+
+First add the `setupLazy` method to the directive. It should set `didSetupLazy`
+so that it's only called once and then call `SimpleTooltip.lazy`. The call to
+`lazy` will store the `tooltip` on the directive and then call the
+`renderTooltipContent` method that you'll implement next.
+
+```ts
+setupLazy() {
+  this.didSetupLazy = true;
+  SimpleTooltip.lazy(this.part!.element, (tooltip: SimpleTooltip) => {
+    this.tooltip = tooltip;
+    this.renderTooltipContent();
+  });
+}
+```
+
+Now add the `renderTooltipContent` method to the directive. It will call Lit's
+`render` function to render the `tooltipContent` into the tooltip element.
+You'll use the part's `renderOptions` so they stay in sync.
+
+```ts
+renderTooltipContent() {
+  render(this.tooltipContent, this.tooltip!, this.part!.options);
```

Comment:
lit's render, since it's not used will be elided by the TS compiler. Make sure to either have a dummy call to it or to include in this snippet to import the render method.

Also you seem to have two imports to 'lit'

Reply:
Done